src/any_agent/callbacks/base.py

```diff
@@ -0,0 +1,28 @@
+# mypy: disable-error-code="no-untyped-def"
+from __future__ import annotations
+
+from typing import Any
+
+
+class Callback:
+    """Base class for all callbacks."""
+
+    def before_llm_call(
+        self, context: dict[str, Any], *args, **kwargs
```

Comment:
I see a lot of places for params: the context, args and kwargs. What would be the guidelines to place what information where? Can it be simplified?